### PR TITLE
PWN-8133 - fix fees in bottomsheet logic

### PR DESCRIPTION
**Pull request by kabramovp2p**
## Jira Ticket

https://p2pvalidator.atlassian.net/browse/PWN-8133

## Description of Work

PWN-8133 - fix fees in bottomsheet logic + resultAmount handling

### CODE REVIEW

File: app/src/main/java/org/p2p/wallet/bridge/send/ui/BridgeSendPresenter.kt [hunk 1]
```diff
@@ -108,7 +108,9 @@ class BridgeSendPresenter(
                 handleUpdateFee(sendFee = state.fee, isInputEmpty = false)
 
                 val textResId = R.string.send_format
-                val value = "${state.amount.formatToken(token.decimals)} ${token.tokenSymbol}"
+                val result = state.bridgeFee?.fee?.resultAmount?.amountInToken
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/bridge/send/ui/BridgeSendPresenter.kt:111]: resultTokenAmount
result is too ambiguous

File: app/src/main/java/org/p2p/wallet/bridge/send/ui/dialog/BridgeSendFeeBottomSheet.kt [hunk 2]
```diff
@@ -84,15 +86,49 @@ class BridgeSendFeeBottomSheet : BaseDoneBottomSheet() {
...
+
     private fun ItemClaimDetailsPartBinding.bindDetailsLineWithFee(title: String, fee: BridgeAmount) {
         textViewTitle.text = title
         textViewFiatAmount.text = fee.fiatAmount?.asUsd() ?: getString(R.string.bridge_info_transaction_free)
-        val formattedTokenAmount = fee.formattedTokenAmount
-        if (formattedTokenAmount == null) {
+        bindFreeOrValue(fee.formattedTokenAmount)
+    }
+
+    private fun ItemClaimDetailsPartBinding.bindDetailsTotalFee(title: String, fees: List<BridgeAmount>) {
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/bridge/send/ui/dialog/BridgeSendFeeBottomSheet.kt:101]: looks like logic that should be in presenter
view should be stupid as possible

File: app/src/main/java/org/p2p/wallet/bridge/send/ui/dialog/BridgeSendFeeBottomSheet.kt [hunk 3]
```diff
@@ -84,15 +86,49 @@ class BridgeSendFeeBottomSheet : BaseDoneBottomSheet() {
...
+        val isFree = fees.isEmpty()
+        val totalFeesByToken = fees.groupBy { it.tokenSymbol }.values.map { bridgeFees ->
+            val feeForTokenData = bridgeFees.firstOrNull()
+            BridgeAmount(
+                tokenSymbol = feeForTokenData?.tokenSymbol.orEmpty(),
+                tokenDecimals = feeForTokenData?.tokenDecimals.orZero(),
+                tokenAmount = bridgeFees.sumOf { it.tokenAmount.orZero() },
+                fiatAmount = bridgeFees.sumOf { it.tokenAmount.orZero() }
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/bridge/send/ui/dialog/BridgeSendFeeBottomSheet.kt:109]: is it okay that you count tokenAmount here?

File: app/src/main/java/org/p2p/wallet/bridge/send/ui/dialog/BridgeSendFeeBottomSheet.kt [hunk 4]
```diff
@@ -84,15 +86,49 @@ class BridgeSendFeeBottomSheet : BaseDoneBottomSheet() {
...
+            )
+        }
+
+        textViewTitle.text = title
+        textViewFiatAmount.text = totalFeesByToken.takeUnless { it.all { isFree } }
+            ?.sumOf { it.fiatAmount.orZero() }
+            ?.asUsd() ?: getString(R.string.bridge_info_transaction_free)
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/bridge/send/ui/dialog/BridgeSendFeeBottomSheet.kt:116]:
```suggestion
            ?.asUsd() 
            ?: getString(R.string.bridge_info_transaction_free)
```

File: app/src/main/java/org/p2p/wallet/bridge/send/ui/dialog/BridgeSendFeeBottomSheet.kt [hunk 5]
```diff
@@ -84,15 +86,49 @@ class BridgeSendFeeBottomSheet : BaseDoneBottomSheet() {
...
+        val formattedTokenAmount = buildString {
+            totalFeesByToken.forEach {
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/bridge/send/ui/dialog/BridgeSendFeeBottomSheet.kt:118]: have you tried .joinToString?

File: app/src/main/java/org/p2p/wallet/bridge/send/ui/mapper/BridgeSendUiMapper.kt [hunk 6]
```diff
@@ -23,20 +23,19 @@ class BridgeSendUiMapper(private val resources: Resources) {
...
-        val resultAmount = getResultAmount(tokenSymbol, decimals, calculationMode, fees)
         return BridgeFeeDetails(
             recipientAddress = recipientAddress,
-            willGetAmount = resultAmount.toBridgeAmount(),
+            willGetAmount = fees?.resultAmount.toBridgeAmount(),
             networkFee = fees?.networkFee.toBridgeAmount(),
             messageAccountRent = fees?.messageAccountRent.toBridgeAmount(),
             bridgeFee = fees?.arbiterFee.toBridgeAmount(),
-            total = resultAmount.toBridgeAmount()
+            totalFees = fees?.let { it ->
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/bridge/send/ui/mapper/BridgeSendUiMapper.kt:34]: better to extract to variable to keep creating BridgeFeeDetails clean

**Review by gslevinkov — APPROVED** [review]

File: app/src/main/java/org/p2p/wallet/bridge/send/ui/dialog/BridgeSendFeeBottomSheet.kt (hunk 2 above)
Comment (kabramovp2p) [reply app/src/main/java/org/p2p/wallet/bridge/send/ui/dialog/BridgeSendFeeBottomSheet.kt:101]: hmmm you are right will move it

File: app/src/main/java/org/p2p/wallet/bridge/send/ui/dialog/BridgeSendFeeBottomSheet.kt (hunk 3 above)
Comment (kabramovp2p) [reply app/src/main/java/org/p2p/wallet/bridge/send/ui/dialog/BridgeSendFeeBottomSheet.kt:109]: no
will fix it thanks

**Review by Davran331122 — APPROVED** [review]

**Pull request merged**